### DISCUSSION TITLE
Query for all switch status in a multi relay strip

### DISCUSSION THREAD

**Discussion by mvadu** [arendst/Tasmota#12265] [Q&A] [answered]
I am using a [geeni 6 outlet strip](https://templates.blakadder.com/geeni_GN-SW003-199S.html) . I want to query current status of all switch positions when my nodered flow starts. I followed the documentation (https://tasmota.github.io/docs/MQTT/) which status "cmnd/tasmota_switch/Power ←     // an empty message/payload sends a status query". But the in my case its only responding with first switch status. 

```
05:31:28.402 CMD: geeni/cmnd/Power
05:31:28.408 MQT: geeni/stat/RESULT = {"POWER1":"ON"}
05:31:28.412 MQT: geeni/stat/POWER1 = ON
```

How can I query for all status (other than sending 6 commands iterating the POWER1->POWER6?

**Comment by barbudor** [comment, marked as answer]
You can query all power states with `POWER0`

```
08:20:25.818 CMD: power0
08:20:25.825 MQT: stat/skt-pcvalou/POWER1 = {"POWER1":"ON"}
08:20:25.830 MQT: stat/skt-pcvalou/POWER1 = ON (retained)
08:20:25.835 MQT: stat/skt-pcvalou/POWER2 = {"POWER2":"OFF"}
08:20:25.840 MQT: stat/skt-pcvalou/POWER2 = OFF (retained)
08:20:25.845 MQT: stat/skt-pcvalou/POWER3 = {"POWER3":"OFF"}
08:20:25.850 MQT: stat/skt-pcvalou/POWER3 = OFF (retained)
08:20:25.855 MQT: stat/skt-pcvalou/POWER4 = {"POWER4":"OFF"}
08:20:25.860 MQT: stat/skt-pcvalou/POWER4 = OFF (retained)
```

But it would be simplest to use `PowerRetain 1` so your broker will remember the last state of POWER*n* as published by Tasmota.
When subscribing to the topic, NodeRed will receive the last valid value without having to actively request it.
Exemple, subscribing using a command line tool (would be the same from NodeRed) on POWER3 immediately gets the retained value:
```
$ msub -t "stat/pc-strip1/POWER3"
2021-06-02 08:23:37: stat/pc-strip1/POWER3 OFF {qos:0, retain:1}
```

**Comment by mvadu** [comment]
Thank you for the neat tip.. I think it needs to be in the documentation. In my case I have a single subscriber for all slots, and needed a way to get series of messages on startup. This is my working solution, with the inject sending the POWER0 at start up.

![image](https://user-images.githubusercontent.com/8618235/120507823-25674380-c395-11eb-9194-32a5b0bf82fe.png)

**Reply by sfromis** [reply]
The command `Power0` is already listed among available commands, in the [Control](https://tasmota.github.io/docs/Commands/#control) section.
